public/views/partials/lib/flash/pop.liquid

```diff
@@ -0,0 +1,14 @@
+{% comment %}
+  returns and reset flash messages
+{% endcomment %}
+{% liquid
+  if context.session.sflash != blank
+    assign flash = context.session.sflash | parse_json
+    if context.location.pathname != flash.from or flash.force_clear
```

Comment:
just a note - the first condition can cause some problems if you redirect to the same path as a form ; for example we do it in `/profile/edit`, after successful submission we redirect to `/profile/edit` as well :)

Reply:
@nemvalid I've tried to implement `sesssion/set` with dynamic `key` but it is not possible now. `session` tag do not allow to use variable as a key name. Is works like the `assign` tag.

Reply:
There might be a way to handle this via mutation.....